File: frontend/scripts/generate-app-icons.py
"""Wide logo -> square PNG icons for Expo/Android APK."""
from pathlib import Path

try:
    from PIL import Image
except ImportError:
    print("Pillow kerak: pip install Pillow")
    raise SystemExit(1)

ROOT = Path(__file__).resolve().parent.parent
SRC = ROOT / "assets" / "images" / "logo-brand.png"
OUT = ROOT / "assets" / "images"
SIZE = 1024
PAD = int(SIZE * 0.12)
INNER = SIZE - PAD * 2

img = Image.open(SRC).convert("RGBA")
img.thumbnail((INNER, INNER), Image.Resampling.LANCZOS)
canvas = Image.new("RGBA", (SIZE, SIZE), (255, 255, 255, 255))
x = (SIZE - img.width) // 2
y = (SIZE - img.height) // 2
canvas.paste(img, (x, y), img)
rgb = Image.new("RGB", (SIZE, SIZE), (255, 255, 255))
rgb.paste(canvas, mask=canvas.split()[3])

for name in ("icon.png", "adaptive-icon.png", "splash-icon.png", "favicon.png"):
    out = OUT / name
    rgb.save(out, "PNG", optimize=True)
    print("Yozildi:", out.name)

print("Tayyor.")

File: frontend/scripts/generate-app-icons.mjs
/**
 * Wide logo -> square app icons (Android adaptive icon safe zone).
 * Run: node scripts/generate-app-icons.mjs
 */
import { existsSync } from "fs";
import { dirname, join } from "path";
import { fileURLToPath } from "url";

const __dirname = dirname(fileURLToPath(import.meta.url));
const root = join(__dirname, "..");
const src = join(root, "assets/images/logo-brand.png");
const outDir = join(root, "assets/images");

async function main() {
  let sharp;
  try {
    sharp = (await import("sharp")).default;
  } catch {
    console.error("sharp o'rnatilmagan. npm install sharp --save-dev");
    process.exit(1);
  }
  if (!existsSync(src)) {
    console.error("logo-brand.png topilmadi:", src);
    process.exit(1);
  }

  const size = 1024;
  const padding = Math.round(size * 0.12);
  const inner = size - padding * 2;

  const logo = sharp(src).resize(inner, inner, { fit: "contain", background: { r: 255, g: 255, b: 255, alpha: 1 } });
  const logoBuf = await logo.png().toBuffer();

  const square = await sharp({
    create: {
      width: size,
      height: size,
      channels: 4,
      background: { r: 255, g: 255, b: 255, alpha: 1 },
    },
  })
    .composite([{ input: logoBuf, gravity: "center" }])
    .png()
    .toBuffer();

  const names = ["icon.png", "adaptive-icon.png", "splash-icon.png", "favicon.png"];
  for (const name of names) {
    await sharp(square).toFile(join(outDir, name));
    console.log("Yozildi:", name);
  }
  console.log("Tayyor — kvadrat ikonlar (1024x1024, markazda logo).");
}

main().catch((e) => {
  console.error(e);
  process.exit(1);
});
